ui/js/controllers/bugfiler.js

```diff
@@ -0,0 +1,245 @@
+"use strict";
+
+treeherder.controller('BugFilerCtrl', [
+    '$scope', '$rootScope', '$modalInstance', '$http', 'summary', 'thBugzillaProductObject',
+    'thPinboard', 'thEvents', 'fullLog', 'parsedLog', 'reftest', 'selectedJob', 'allFailures',
+    'thNotify', 'ThLog',
+    function BugFilerCtrl(
+        $scope, $rootScope, $modalInstance, $http, summary, thBugzillaProductObject,
+        thPinboard, thEvents, fullLog, parsedLog, reftest, selectedJob, allFailures,
+        thNotify, ThLog) {
+
+        var $log = new ThLog("BugFilerCtrl");
+
+        /**
+         *  'enter' from the product search input should initiate the search
+         */
+        $scope.productSearch = function(ev) {
+            if (ev.keyCode === 13) {
+                $scope.findProduct();
+            }
+        };
+
+        /*
+         **
+         */
+        $scope.isReftest = function() {
+            return reftest !== "";
+        };
+
+        /**
+         *  Pre-fill the form with information/metadata from the failure
+         */
+        $scope.initiate = function() {
+            $modalInstance.parsedSummary = $modalInstance.parseSummary(summary);
+
+            $scope.modalSummary = "Intermittent " + $modalInstance.parsedSummary[0].join(" | ");
+
+            $("#modalParsedLog").next().attr("href", parsedLog);
+            $("#modalFullLog").next().attr("href", fullLog);
+            if ($scope.isReftest()) {
+                $("#modalReftestLog").next().attr("href", reftest);
+            }
+
+            var thisFailure = "";
+            for(var i=0;i<allFailures.length;i++) {
```

Comment:
we usually use spacing like:

    for(var i = 0; i < allFailures.length; i++) {